python/beeai_framework/agents/tool_calling/agent.py

```diff
@@ -213,3 +213,11 @@ def _generate_templates(
             else:
                 setattr(templates, name, override(getattr(templates, name)))
         return templates
+
+    async def clone(self) -> "ToolCallingAgent":
+        cloned = ToolCallingAgent(
+            llm=self._llm, memory=self._memory, tools=self._tools, templates=self._templates.model_dump()
+        )
+        cloned.emitter = await self.emitter.clone()
```

Comment:
This might break stuff if some extra operations are defined in the constructor. For instance, if the class attaches a handler in the constructor, you override it here.

Reply:
to clarify are you referring to the `listeners` in emitter?

Reply:
yes

Reply:
https://github.com/i-am-bee/beeai-framework/pull/705/commits/73e0d8e6130e9a39d543164287a33bb60bbd928d

Reply:
What about others? I don't see copy of memory, tools etc.

Reply:
https://github.com/i-am-bee/beeai-framework/pull/705/commits/dc26f41f4a3ddb1a6a32e67d83220b9b12c73d70